tests/regression/test_voting_acc.py

```diff
@@ -121,3 +121,99 @@ def test_can_object(stranger, test_vote):
+    # stranger vote for YES holders 1,2 (reject no delegation)
+    with reverts("REJECT"):
+        contracts.voting.attemptVoteForMultiple(vote_id, True, [holder1, holder2], {"from": stranger})
+
+    # holders 0,1 delegate to stranger
+    for holder in [holder1, holder2]:
+        contracts.voting.setDelegate(stranger, {"from": accounts.at(holder, force=True)})
+
+    # stranger vote for YES holder 3 (reject not a delegation for holder 3)
+    with reverts("REJECT"):
+        contracts.voting.attemptVoteForMultiple(vote_id, True, [holder3], {"from": stranger})
+
+    # stranger vote for YES holders 1,2,3 (reject not a delegation for holder 3)
+    with reverts("REJECT"):
+        contracts.voting.attemptVoteForMultiple(vote_id, True, [holder1, holder2, holder3], {"from": stranger})
+
+    #  stranger revote for NO holders 1,2
+    contracts.voting.attemptVoteForMultiple(vote_id, True, [holder1, holder2], {"from": stranger})
+
+    # holder 3 delegate to stranger
+    contracts.voting.setDelegate(stranger, {"from": accounts.at(holder3, force=True)})
+
+    holders = contracts.voting.getDelegatedVoters(stranger, 0 , 5)
+    assert holders == [holder1, holder2, holder3]
+
+    # stranger vote for YES holders 3
+    contracts.voting.attemptVoteForMultiple(vote_id, True, [holder3], {"from": stranger})
+
+    #  holder 3 vote for NO
+    contracts.voting.vote(vote_id, False, False, {"from": accounts.at(holder3, force=True)})
+
+    #  holder 2 delegate to holder 3
+    contracts.voting.setDelegate(holder3, {"from": accounts.at(holder2, force=True)})
+
+    #  holder 1 reset delegate
+    contracts.voting.resetDelegate({"from": accounts.at(holder1, force=True)})
+
+    # stranger revote for YES holders 1-3 (reject)
+    with reverts("REJECT"):
+        contracts.voting.attemptVoteForMultiple(vote_id, True, [holder1, holder2, holder3], {"from": stranger})
+
+    holders = contracts.voting.getDelegatedVoters(stranger, 0 , 5)
+    assert holders == [holder3]
+
+    # stranger revote for YES holders 1 (reject not a delegate)
+    with reverts("REJECT"):
+        contracts.voting.attemptVoteForMultiple(vote_id, True, [holder1], {"from": stranger})
+
+    # stranger revote for YES holders 2 (reject not a delegate)
+    with reverts("REJECT"):
+       contracts.voting.attemptVoteForMultiple(vote_id, True, [holder2], {"from": stranger})
+
+    # stranger revote for YES holders 3 (reject cant overpower holder)
+    with reverts("REJECT"):
+        contracts.voting.attemptVoteForMultiple(vote_id, True, [holder3], {"from": stranger})
+
+    # holders 1,2,3 can revote main phase YES
+    for holder in [holder1, holder2, holder3]:
+        contracts.voting.vote(vote_id, True, False, {"from": accounts.at(holder, force=True)})
+
+    chain.sleep(contracts.voting.voteTime() - contracts.voting.objectionPhaseTime())
+    chain.mine()
+
+    assert contracts.voting.getVotePhase(vote_id) == 1  # Objection phase
```

Comment:
Let's also add the check that the delegate can't vote "yea" during the objection phase.